app/controllers/registers_controller.rb

```diff
@@ -108,42 +108,38 @@ def recover_records(register_id, fields, params, page_size = 100)
       has_name_field ? 'name' : params[:id]
     }
 
-    literal = params[:q]
+    search_term = params[:q]
     status = params[:status] ||= 'current'
     page = params[:page] ||= 1
     sort_by = params[:sort_by] ||= default_sort_by.call
     sort_direction = params[:sort_direction] ||= 'asc'
 
     query = Record.where(spina_register_id: register_id, entry_type: 'user')
-    count_query = Record.select(1).where(spina_register_id: register_id, entry_type: 'user')
 
     case status
     when 'archived'
       query = query.where("data->> 'end-date' is not null")
-      count_query = query.where("data->> 'end-date' is not null")
     when 'current'
       query = query.where("data->> 'end-date' is null")
-      count_query = query.where("data->> 'end-date' is null")
     end
```

Comment:
The conventional way to make this is putting the assignment outside the case:

```ruby
query = case status
when 'archived'
  query.where("data->> 'end-date' is not null")
when 'current'
  query.where("data->> 'end-date' is null")
end
```

Also, what happens if `status` is neither "archived" nor "current"?

Reply:
Ah makes sense 👍